packages/network/src/plane/bus_impl.rs

```diff
@@ -223,3 +233,228 @@ where
         Some(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        msg::TransportMsg,
+        plane::{
+            bus::{HandleEvent, HandlerRoute, PlaneBus},
+            bus_impl::PlaneBusImpl,
+            NetworkPlaneInternalEvent,
+        },
+        transport::{ConnectionSender, MockConnectionSender},
+    };
+    use async_std::channel::{unbounded, TryRecvError};
+    use bluesea_identity::{ConnId, NodeAddr, NodeId};
+    use bluesea_router::{MockRouterTable, RouteAction, RouteRule};
+    use std::sync::Arc;
+
+    type HE = ();
+    type BE = ();
+
+    fn create_mock_connection(conn: ConnId, node: NodeId, addr: NodeAddr) -> MockConnectionSender {
+        let mut sender = MockConnectionSender::new();
+        sender.expect_conn_id().return_const(conn);
+        sender.expect_remote_node_id().return_const(node);
+        sender.expect_remote_addr().return_const(addr);
+        sender.expect_send().return_const(());
+
+        sender
+    }
+
+    #[test]
+    fn should_add_remove_close_connection() {
+        let local_node_id = 1;
+        let (plane_tx, _plane_rx) = unbounded();
+        let router = Arc::new(MockRouterTable::new());
+        let bus = PlaneBusImpl::<BE, HE>::new(local_node_id, router, plane_tx);
+        let mut sender = create_mock_connection(ConnId::from_in(1, 1), 2u32, NodeAddr::empty());
+        sender.expect_close().times(1).return_const(());
+        let conn = Arc::new(sender);
+        let conn_c = conn.clone();
+        let rx = bus.add_conn(conn.clone());
+        assert!(rx.is_some());
+
+        // duplicate conn
+        assert!(bus.add_conn(conn).is_none());
```

Comment:
should be assert_eq!(bus.add_conn(conn), None);

Reply:
bus.add_conn(conn) returns a Receiver, which cannot be assert_eq!